ocaml/xapi/message_forwarding.ml

```diff
@@ -449,6 +449,27 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
         Ref.string_of sdn_controller
     with _ -> "invalid"
 
+  let pusb_uuid ~__context pusb =
+    try if Pool_role.is_master () then
+        Db.PUSB.get_uuid __context pusb
+      else
+        Ref.string_of pusb
+    with _ -> "invalid"
+
+  let usb_group_uuid ~__context usb_group =
+    try if Pool_role.is_master () then
+        Db.USB_group.get_uuid __context usb_group
+      else
+        Ref.string_of usb_group
+    with _ -> "invalid"
+
+  let vusb_uuid ~__context vusb =
+    try if Pool_role.is_master () then
+        Db.VUSB.get_uuid __context vusb
+      else
+        Ref.string_of vusb
+    with _ -> "invalid"
+
```

Comment:
What's this for? If nothing else, it's badly named (Ref.string_of doesn't give you a uuid) - and message_forwarding code is only executed on the master, so the Pool_role.is_master is unnecessary.

Reply:
As seems  to keep consistency with existing style of other implementation(eg. pgpu_uuid gpu_group_uuid .... )

Reply:
Yuck, yes. We need to redo this set of functions. Let's not hold up your PR for that though :-)